system/jlib/jstats.cpp

```diff
@@ -2091,7 +2091,35 @@ class TreeNodeStatisticGatherer : public CInterfaceOf<IStatisticGatherer>
 
 void CRuntimeStatistic::merge(unsigned __int64 otherValue, StatsMergeAction mergeAction)
 {
-    value = mergeStatisticValue(value, otherValue, mergeAction);
+    switch (mergeAction)
+    {
+    case StatsMergeKeepNonZero:
+        if (otherValue)
+        {
+            unsigned __int64 zero = 0;
+            value.compare_exchange_weak(zero, otherValue);
```

Comment:
Should be compare_exchange_strong.   compare_exchange_weak is for use in loops (details of the difference are connected to how how cas is implemented on non intel architectures).